Project 5 For John Beverly

```diff
@@ -59,345 +59,388 @@ Group 2 contains Emanuel, Katie, Josh V (commonly know as V by group member), Ca
 
-sh:path ex:spatialcontainment ;              
+sh:maxCount 1 ;    
 
-sh:datatype xsd:anyURI ;                
+sh:datatype xsd:string ; 
 
-sh:in (ex:city ex:country ex:continent) ;          
+] .
 
-sh:message "Location must be within the specified spatial boundary" ;              
+5.	Constraint for Irreflexive 
+@prefix sh: <http://www.w3.org/ns/shacl#> .
 
-sh:minCount 1 ;           
+@prefix ex: <http://example.com/> .
 
-sh:maxCount 1 ;           
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
 
-sh:name "Location" ;           
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+ex:IrreflexiveShape
+
+a sh:NodeShape ;
+
+sh:targetClass ex:relation ;
+
+sh:property [    
 
-sh:pattern "ex:city|ex:country|ex:continent" ;     
+sh:path rdf:value ;    
+
+sh:minCount 1 ;    
+
+sh:maxCount 1 ;    
+
+sh:datatype xsd:string ; 
 
 ] .
 
 
-3rd Constraint Begins Below 
 
+6.	Constraint for Transitivity 
 @prefix sh: <http://www.w3.org/ns/shacl#> .
 
-@prefix ex: <http://example.org/> .
+@prefix ex: <http://example.com/> .
+
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
 
 @prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
 
 
-ex:PersonShape 
+ex:TransitiveShape
 
 a sh:NodeShape ;
 
-sh:targetClass ex:Person ;
+sh:targetClass ex:relation ;
 
-sh:property [
+sh:property [    
+
+sh:path rdf:value ;    
 
-sh:path ex:gender ;
+sh:minCount 1 ;    
 
-sh:in (ex:Male ex:Female ex:Nonbinary) ;
+sh:maxCount 1 ;    
 
-sh:message "Gender must be Male, Female, or Nonbinary." ;
+sh:datatype xsd:string ; 
 
 ] .
 
+7.	Constraint for Transitivity and Symmetry 
+@prefix sh: <http://www.w3.org/ns/shacl#> .
 
-ex:GenderConstraint 
+@prefix ex: <http://example.com/> .
 
-a sh:NodeShape ;
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
 
-sh:targetClass ex:Gender ;
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
 
-sh:property [
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+ex:TransSymmetryShape
+
+a sh:NodeShape ;
 
-sh:path ex:identifiesAs ;
+sh:targetClass ex:relation ;
 
-sh:in (ex:Male ex:Female ex:Nonbinary) ;
+sh:property [    
+
+sh:path ex:hasSource ;    
 
-sh:message "Gender identity must be Male, Female, or Nonbinary." ;
+sh:class ex:Node  
 
 ] ;
 
-sh:property [
+sh:property [    
 
-sh:path ex:mostOfTheTimeIdentifiesAs ;
+sh:path ex:hasTarget ;    
 
-sh:in (ex:Male ex:Female ex:Nonbinary) ;
+sh:class ex:Node  ] ;
 
-sh:message "Most of the time gender identity must be Male, Female, or Nonbinary." ;
+sh:property [    
 
-] ;
+sh:path ex:hasValue ;    
 
-sh:property [
+sh:or (       [ sh:datatype xsd:string ; 
 
-sh:path ex:sometimesIdentifiesAs ;
+sh:in ( "same, opposite" ) 
 
-sh:in (ex:Male ex:Female ex:Nonbinary) ;
+]
 
-sh:message "Sometimes gender identity must be Male, Female, or Nonbinary." ;
+[ sh:datatype xsd:boolean ]
 
-] .
+)
 
+] ;
+
+sh:property [    
 
-4th Constraint Begins Below
+sh:path ex:hasType ;    
 
+sh:in ( "symmetric, transitive" )  
```

Comment:
If I understand this correctly, you are saying that things are either strings of type same/opposite or booleans? I am not sure how this constrains symmetry at a structural level, outside of a specific schema that might be using these specific properties and values

Reply:
Second that